cbqueryx/queryrespreader.go

```diff
@@ -155,7 +155,11 @@ func (r *queryRespReader) parseError(errJson *queryErrorJson) *ServerError {
 		err = ErrIndexExists
 	}
 	if errCode == 12003 {
-		err = ErrCollectionNotFound
+		fields := strings.Fields(errJson.Msg)
+		err = ResourceError{
+			Cause:        ErrCollectionNotFound,
+			ResourcePath: fields[len(fields)-1],
```

Comment:
I think it might be worth adding a comment somewhere here just illustrating what the msg looks like for this, so that it's clear what len(fields) - 1 is grabbing. Do we have tests somewhere for checking resource path on these errors? I'm thinking of the scenario where query changes the error message, and we want to detect that quickly.